portal-test/gtk4/main.js

```diff
@@ -0,0 +1,64 @@
+
+pkg.initGettext();
+pkg.initFormat();
+pkg.require({
+    'GdkPixbuf': '2.0',
+    'Gio': '2.0',
+    'Gst': '1.0',
+    'Gtk': '4.0',
+    'Xdp': '1.0',
+    'XdpGtk4': '1.0',
+});
+
+const { Gdk, Gio, GLib, Gst, Gtk } = imports.gi;
+
+const { Application } = imports.application;
+
+function stopRunningInstance() {
+    const bus = Gio.DBus.session;
+
+    const mainloop = GLib.MainLoop.new(null, false);
+
+    const watchId = bus.watch_name(
+        'org.gnome.PortalTest.Gtk4',
+        0,
+        (bus, name, owner) => {
+            log(`Name ${name} owned by ${owner}`);
+        },
+        (bus, name) => {
+            log(`Name ${name} unowned`);
+            mainloop.quit();
+        });
+
+    try {
+        bus.call_sync(
+            'org.gnome.PortalTest.Gtk4',
+            'org.gnome.PortalTest.Gtk4',
+            'org.gtk.Actions',
+            // TODO
```

Comment:
Empty TODO left here

Reply:
Uh, I forgot about these. Implemented and tested this code path now (there was another TODO in `application.js` which was fixed too)